cody.hh

```diff
@@ -577,7 +577,7 @@ public:
 class Server
 {
 public:
-  enum Direction {READING, WRITING, PROCESSING};
+  enum Direction {READING, WRITING, PROCESSING, STALLED};
```

Comment:
This isn't right.  Is it for the case when we're waiting on make to do something before we can assemble a response?
If so, that is what the PROCESSING state is for.  I only expected the server to need 3 states.

Reply:
Yep, that was the primary reason I added another state. I was also using that extra state to track if ProcessRequests() was resolving an InvokeSubProcessRequest, so I know not to call PrepareToWrite() immediately after ProcessRequests(). 

I thought about adding a state member to my derived Cody::Server class. I originally wanted to pass in a derived Cody::Server class instance to InvokeSubProcessRequest, but the compiler kept complaining that the virtual function override expects a Cody::Server class parameter. Any advice?

Reply:
you need that information in the client_state object, which IIRC either derives from or has Cody::Server as a member.